src/leader_election.rs

```diff
@@ -251,7 +273,14 @@ pub mod ballot_leader_election {
         }
 
         fn hb_timeout(&mut self) -> Option<Leader<Ballot>> {
+            info!(self.logger, "Heartbeat timeout");
+
             let result: Option<Leader<Ballot>> = if self.ballots.len() + 1 >= self.majority {
```

Comment:
I think it makes sense to have a `warn!` that logs `self.ballots` when we don't receive a majority